olga/analytics/utils.py

```diff
@@ -1,133 +1,30 @@
+from django.http import HttpResponse
 
-def probability(x):
-    return random() < x
+from .forms import EdxInstallationForm, InstallationStatisticsForm
 
+HTTP_401_UNAUTHORIZED = 401
 
-def generate_model():
 
-    cl = LoadTestModel()
-    # for middle cluster
-
-    instance_list = []
-    number_of_instances = cl.clusters['middle']['quantity']
-
-    instance_dict = {}
-    students_dict = {}
-
-    for instance_num in range(number_of_instances):
-        for day in range(cl.days):
-            if probability(cl.send_data):
-                dt = (datetime.datetime.today() - datetime.timedelta(days=cl.days - day)).strftime('%Y-%m-%dT%H:%M:%SZ')
-
-                if cl.probability_to_add_course:
-                    instance_dict[instance_num] = instance_dict.get(instance_num, 0) + 1
-                    students_dict[instance_num] = students_dict.get(instance_num, 0) + 25
-
-                courses = instance_dict.get(instance_num, 0)
-
-                if cl.probability_to_add_student:
-                    students_dict[instance_num] = students_dict.get(instance_num, 0) + 0
-
-                students = students_dict.get(instance_num, 0)
-
-                pk = instance_num * cl.days + day
-
-                edx_installation = {
-                    "model": "analytics.edxinstallation",
-                    "pk": pk,
-                    "fields": {
-                        "access_token": "platform" + " " + str(instance_num),
-                        "latitude": 1.0,
-                        "longitude": 1.0,
-                        "platform_name": "test",
-                        "platform_url": "test",
-                    }
-                }
-
-                installation_statistics = {
-                    "model": "analytics.installationstatistics",
-                    "pk": pk,
-                    "fields": {
-                        "active_students_amount_day": students,
-                        "active_students_amount_week": 1000,
-                        "active_students_amount_month": 1000,
-                        "courses_amount": courses,
-                        "data_created_datetime": dt,
-                        "edx_installation": pk,
-                        "statistics_level": 1,
-                        "students_per_country": "{\"RU\": 2632, \"CA\": 18543, \"UA\": 2011, \"null\": 1}"
-                    }
-                }
+def installation_statistics_forms_checker(receive_installation_statistics_post_method):  # pylint: disable=invalid-name
+    """
+    Checksare edX overall installation info and statistics valid with corresponding forms.
 
-                instance_list.append(installation_statistics)
-                instance_list.append(edx_installation)
+    Returns HTTP-response with status 401, that means at least one of two forms is not valid.
+    """
+    def receive_installation_statistics_post_method_wrapper(request, *args, **kwargs):  # pylint: disable=invalid-name
+        """
+        Wrapper.
```

Comment:
Why? I had some more text in docstring, but @z4y4ts said me, that simple `Wrapper.` is better.